### PR TITLE
Update zstd to 1.4.3

### PR DESCRIPTION
**Pull request by jmannanc**

### CODE REVIEW

File: core/SCsub [hunk 1]
```diff
@@ -120,6 +120,8 @@ if env['builtin_zstd']:
         "compress/zstd_ldm.c",
         "compress/zstd_opt.c",
         "compress/zstdmt_compress.c",
+		"compress/zstd_compress_literals.c",
```
Comment (akien-mga) [inline core/SCsub]: Indentation issue (mix of tabs and spaces).

Comment (SneakyFish5) [reply core/SCsub]: I think this is GitHub UI bug as I saw this too, but when I actually when into the file to view it, the indentation was normal.

Comment (akien-mga) [reply core/SCsub]: No, indentation is about characters, not visuals. The indentation here should be 8 spaces, but you used 4 spaces + 1 tab. If you configure your tab display to something different from 4 characters width, you'll see the problem.

Comment (SneakyFish5) [reply core/SCsub]: Oh I see, the problem was with my editor then. Fixed it now, sorry about that.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Looks like we can even upgrade to 1.4.4 already: https://github.com/facebook/zstd/releases/tag/v1.4.4

**Comment by SneakyFish5** [comment]
Yup, once I fix what you commented on for opus I'll work on syncing zstd, since this seems like it was a major update.

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**